src/renderer/wg_engine/tvgWgGeometry.cpp

```diff
@@ -458,7 +459,8 @@ void WgGeometryData::appendStroke(const WgPolyline* polyline, const RenderStroke
                     } else if (stroke->join == StrokeJoin::Miter) {
                         WgPoint nrm = (nrm0 + nrm1).normal();
                         float cosine = nrm.dot(nrm0);
-                        if ((cosine != 0.0f) && (abs(cosine) < 1.0f) && (abs(wdt / cosine) <= stroke->miterlimit * 2)) {
+                        float miterRatio = 1.0f / std::sin((std::acos(dir0.dot(dir1.negative())) * 0.5));
+                        if ((!mathZero(cosine)) && (abs(cosine) < 1.0f) && (miterRatio <= stroke->miterlimit)) {
```

Comment:
can you please comment on this: `(abs(cosine) < 1.0f)` ? even if it happens that cosine = 1.0001 I don;t see what problem will it cause.